pep-0690.rst

```diff
@@ -312,57 +325,54 @@ import object. When ``modb.foo`` is later referenced, it will also try to
 Python, and at this point will replace the lazy import object at
 ``modb.__dict__["foo"]`` with the actual module ``foo``.
 
-There is one case in which a lazy import object can escape one dictionary (but
-only into another dictionary) without being resolved. To preserve the
-performance of bulk-copy operations like ``dict.update()`` and ``dict.copy()``,
-they do not check for or resolve lazy import objects. However, if the source
-dict has a ``*_LAZY`` lookup kind set that indicates it might contain lazy
-objects, that lookup kind will be passed on to the updated/copied dictionary.
-This still ensures that the lazy import object can't escape into Python code
-without being resolved.
-
-Other "bulk" dictionary lookup methods (such as ``dict.items()``,
-``dict.values()``, etc) will resolve all lazy import objects in the dictionary.
-Since it is uncommon for any of these to be used on a module namespace
-dictionary, the priority here is simplicity of implementation and minimizing the
-overhead on normal non-lazy dictionaries (just one check to see if the
-dictionary has a ``*_LAZY`` lookup kind).
+There are two cases in which a lazy import object can "escape" one dictionary:
+
+* Into another dictionary: to preserve the performance of bulk-copy operations
+  like ``dict.update()`` and ``dict.copy()``, they do not check for or resolve
+  lazy import objects. However, if the source dict has the ``dk_lazy_imports``
+  flag set that indicates it might contain lazy objects, that flag will be
+  passed on to the updated/copied dictionary. This still ensures that the lazy
+  import object can't escape into Python code without being resolved.
+
+* Through the garbage collector: lazy imported objects are still Python objects
+  and live within the garbage collector; as such, they can be collected and seen
+  by means of using ``gc.collect()`` and ``gc.get_objects()``. Lazy objects are
+  not useful but they are also harmless and pose no danger if extracted from the
+  garbage collector in this way.
```

Comment:
Since they can escape via gc, do lazy objects need to be specified?  What will someone see or what can they do with lazy objects if they get one?  Can they resolve the object they stand in for?  Are there any attributes they can query?  Or is the entire API non-public or inaccessible?  It seems like *something* should be said (not here, but perhaps elsewhere in the PEP) about what such captured lazy objects look like, even if it's something like "their API is intended to be opaque and is completely implementation dependent".

Reply:
The object is self contained and a `repr()` of it would show as something like `<lazy_object 'fully.qualified.name'>` it has no attributes and it can't be resolved in any way. I agree we should mention something about this somewhere.